src/ray/raylet/scheduling/local_resource_manager.cc

```diff
@@ -342,26 +342,26 @@ std::optional<syncer::RaySyncMessage> LocalResourceManager::CreateSyncMessage(
 
   resources_data.set_is_draining(IsLocalNodeDraining());
 
-  std::stringstream node_activity;
   for (const auto &iter : last_idle_times_) {
     if (iter.second == absl::nullopt) {
       // If it is a WorkFootprint
       if (iter.first.index() == 0) {
         switch (std::get<WorkFootprint>(iter.first)) {
         case WorkFootprint::NODE_WORKERS:
-          node_activity << " Node currently has leased workers." << std::endl;
+          resources_data.add_node_activity("Busy workers on node.");
           break;
         default:
           UNREACHABLE;
         }
         // If it is a ResourceID
       } else {
-        node_activity << " Resource: " << std::get<ResourceID>(iter.first).Binary()
-                      << " is currently in use." << std::endl;
+        std::stringstream out;
+        out << "Resource: " << std::get<ResourceID>(iter.first).Binary()
```

Comment:
```suggestion
        out << std::get<ResourceID>(iter.first).Binary() << " in use." 
```
?

I am actually not sure if the binary resource id will show up nice here? will it show GPU for ResourceID::GPU?

Reply:
Here is the definition:
```
 std::string Binary() const { return GetMap().Get(id_); }
 ```

And here is what happens when you initialize a custom resource with a custom name:
```
 explicit BaseSchedulingID(const std::string &name) : id_{GetMap().Insert(name)} {}
```

For the usual ones, GetMap gets a Meyer's singleton initialized with the predefined resources:
```
/// Specialization for SchedulingIDTag. Specifically, we populate
/// the singleton map with PredefinedResources.
template <>
inline StringIdMap &BaseSchedulingID<SchedulingIDTag::Resource>::GetMap() {
  static std::unique_ptr<StringIdMap> map{[]() {
    std::unique_ptr<StringIdMap> map(new StringIdMap());
    map->InsertOrDie(kCPU_ResourceLabel, CPU)
        .InsertOrDie(kGPU_ResourceLabel, GPU)
        .InsertOrDie(kObjectStoreMemory_ResourceLabel, OBJECT_STORE_MEM)
        .InsertOrDie(kMemory_ResourceLabel, MEM);
    return map;
  }()};
  return *map;
}
```

So it basically will show the name, except in the case that the user names their resource with some unprintable string. Then, it is not our problem.

Reply:
ah i see - the name of that function "Binary" is then pretty confusing. I would expect it to be actual binary format, which seems to be just "Name" in this case.